src/connector.ts

```diff
@@ -439,22 +439,29 @@ export let cljSession: NReplSession;
 export let cljsSession: NReplSession;
 
 export async function connect(connectSequence: ReplConnectSequence,
-    isAutoConnect: boolean, hostname?: string, port?: string) {
+    isAutoConnect: boolean, maybeHostname?: string, maybePort?: string) {
     const cljsTypeName = projectTypes.getCljsTypeName(connectSequence);
 
     state.analytics().logEvent("REPL", "ConnectInitiated", isAutoConnect ? "auto" : "manual");
     state.analytics().logEvent("REPL", "ConnectInitiated", cljsTypeName).send();
 
-    const portFile = projectTypes.nreplPortFileUri(connectSequence);
-
     state.extensionContext.workspaceState.update('selectedCljsTypeName', cljsTypeName);
     state.extensionContext.workspaceState.update('selectedConnectSequence', connectSequence);
 
     try {
-        let bytes = await vscode.workspace.fs.readFile(portFile);
-        let port = new TextDecoder("utf-8").decode(bytes);
+        let port: string | undefined = maybePort;
+        if (port === undefined) { // when we connect to a repl started externally
+            const portFile = projectTypes.nreplPortFileUri(connectSequence);
+            // Most of the time the repl started externally is of project provenance
+            // and the port is the one in the nrepl file, use it if there is one.
+            // Noting that's not always the case; if the nrepl is from another project
+            // the port guessed from the opened folder is not the one we want, and
+            // it needs to be manually changed to the right one.
```

Comment:
I think the code is clear enough here. Adding the comments makes bit harder to scan the function.

Reply:
Removed.